gke-deploy/core/container/container.go

```diff
@@ -17,17 +17,83 @@ import (
 	"fmt"
 
 	"github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/services"
+	"github.com/google/go-containerregistry/pkg/name"
 )
 
 const (
 	imageDigestFormat = "value(image_summary.digest)"
 )
 
-// GetDigest uses an image string to get its corresponding digest.
-func GetDigest(ctx context.Context, image string, gs services.GcloudService) (string, error) {
-	digest, err := gs.ContainerImagesDescribe(ctx, image, imageDigestFormat)
+// Image is an alias to name.Reference.
+type Image = name.Reference
+
+// ParseImages parses a slice of image strings.
+func ParseImages(ctx context.Context, images []string) ([]Image, error) {
+	var ims []Image
+
+	exists := make(map[string]bool)
+	for _, image := range images {
+		im, err := parseImage(ctx, image)
+		if err != nil {
+			return nil, err
+		}
+
+		imName, err := GetName(ctx, im)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get image name: %v", err)
+		}
+
+		if ok := exists[imName]; ok {
+			return nil, fmt.Errorf("duplicate image name: %q", imName)
+		}
+		exists[imName] = true
+		ims = append(ims, im)
+	}
+
+	return ims, nil
+}
+
+func parseImage(ctx context.Context, image string) (Image, error) {
+	im, err := name.ParseReference(image)
+	if err != nil {
+		return nil, fmt.Errorf("image is invalid: %q", image)
+	}
+	return im, nil
+}
+
+// GetNameFromString gets an image's name, given a string representation of the image.
+func GetNameFromString(ctx context.Context, image string) (string, error) {
```

Comment:
This name could use some work. How about "ParseImageReference"? Something that indicates the input is a string, representing an image reference, and it returns (I think?) the fully-qualified image reference after normalizing it?

Reply:
I added a comment indicating what it does.
"given "gcr.io/my-project/my-app:1.0.0", returns "gcr.io/project/my-app""

Reply:
Also renamed to "ParseImageReference"